src/lib/MetamaskDelegatedAccess.ts

```diff
@@ -0,0 +1,82 @@
+import { Wallet } from "ethers"
+import { Buffer } from 'buffer'
+import { encrypt } from "eth-sig-util"
+import { bufferToHex } from 'ethereumjs-util'
+import {StreamrClient} from "streamr-client"
+
+declare global {
+    interface Window { 
+        ethereum: any
```

Comment:
Could this be `object`?

Reply:
the linter has strong opinions about it:
```
Don't use `Object` as a type. The `Object` type actually means "any non-nullish value", so it is marginally better than `unknown`.
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead
```

type `unknown` doesn't allow for methods/properties to be accesses, `any` seems like the least of the evils here

Reply:
Ah, right you are! While trying to disprove you I bumped into this package, might be worth a look?https://github.com/MetaMask/providers